docs/proposals/2017-04-27 Text Formatting.md

```diff
@@ -0,0 +1,176 @@
+
+* a required `start` and `length` integer properties define sub-string range,
+* an optional `attributes` property with array of `attribute` objects.
+
+Example:
+```
+    "_id": "AAAAAAAAAAAAAAAAAAAAAAAAAA",
+    "text": "The text attribute should contain plain text content only.",
+    "selections": [
+      {
+        "_id": "AAAAAAAAAAAAAAAAAAAAAAAAAB",
+        "type": "selection",
+        "start": 4,
+        "length": 4,
+        "attributes": [ ... ]
+      },
+      { ... }
+    ]
+    "content": "The <b>text</b> attribute should contain <i>plain text</i> content only."
+```
+
+### `attribute`s are modifications applied to the text string or selections
+
+Each attribute element modifies the formatting of the text it is applied to. Initial list of supported attribute types shall be:
+
+* strong,
+* emphasis,
+* strike,
+* highlight,
+* superscript,
+* subscript,
+* smallcaps,
+* dropcaps.
+
+Example:
+```
+        "_id": "AAAAAAAAAAAAAAAAAAAAAAAAAB",
+        "type": "selection",
+        "start": 4,
+        "length": 4,
+        "attributes": [
+          {
+            "_id": "AAAAAAAAAAAAAAAAAAAAAAAAAC",
+            "type": "strong"
+          }
+        ]
+```
+
+
+## Part II. Structured Text Applied to Other Elements
+
+### All formatted content should be represented as a structured text content
+
+Existing `blockquote`, `code`, `correction`, `header`, `interstitial_link` adopt the same structure as a `text` element. Following properties are replaced with text/attributes/selections/text properties:
+
+* blockquote/content
+* code/content
+* correction/text
+* header/content
+* interstitial_link/content
+* text/content
+
+
+## Part III. Document Usage of HTML
+
+### Define properties allowing HTML content
+
+All `string` type properties which can have HTML content should be explicitly documented. These include:
+
+* raw_html/content
+* video/embed_html
+
+All other `string` type properties are assumed to have only plain text content. Examples include:
+
+* image/subtitle
+* image/caption
+* code/language
+* address/street_address (via trait_address.json)
+* copyright (via trait_copyright.json)
+* editor_note (via trait_editor_note.json)
+* location (via trait_location.json)
+* owner/name (via trait_owner.json)
+* slug (via trait_slug.json)
+* source/name (via trait_source.json)
+* author/name
+* author/org
+* author/bio
+* auxiliary/name (taxonomy)
+* keyword/keyword (taxonomy)
+* named_entity/name (taxonomy)
+* site/name
+* site/description
+* tag/text
+* tag/description
+* topic/name
+* video/rating
+
+
+## Part IV. Future Work
+
+This proposal leaves room for defining other types of attributes, such a `link`, which could be used to denote relationship between elements and other content.
+
+
+# OPEN QUESTIONS #
+
+* Should text (trait_label.json) support formatting/HTML?
+* Should main (train_disctionary.json) support formatting/HTML?
```

Comment:
Arguably these should be dropped and the fields that use them should each define their own schema -- they are more utility groupings for schema than actual semantic traits

But my hot take is:
  - labels: no formatting
  - headlines, subheadlines, description: formatting